src/external/song.ts

```diff
@@ -17,100 +17,186 @@
-  logger.info(`Retrieved ${studies.length} studies: ${studies}`);
-  return studies;
+	const studiesUrl = urljoin(url, '/studies/all');
+	logger.info(`Fetching studies from: ${studiesUrl}`);
+	const res = await fetch(`${url}/studies/all`);
+	const studies = await res.json();
+	logger.info(`Retrieved ${studies.length} studies: ${studies}`);
+	return studies;
 };
 
-/**
- * Fetch potentially very large responses from Song then return them in a batched stream.
- * @param analysesUrl
- * @returns
- */
-const fetchAnalysesInBatches = async (analysesUrl: string): Promise<Batch> => {
-  const controller = new abortController();
-  const timeoutPeriod = (await getAppConfig()).datacenter.fetchTimeout;
-  const batchSize = (await getAppConfig()).datacenter.batchSize;
-  const timeout = setTimeout(() => {
-    logger.warn(`Aborting request for analyses due to timeout after ${timeoutPeriod}`);
-    controller.abort();
-  }, timeoutPeriod);
-  try {
-    logger.info(`Fetching analyses from ${analysesUrl}`);
-    const res = await fetch(analysesUrl, {
-      signal: controller.signal,
-    });
-    const resStream = res.body;
-    const pipeline = resStream.pipe(streamArray.withParser()).pipe(new Batch({ batchSize }));
-    logger.info(`Received analyses response, returning as Batch.`);
-    return pipeline;
-  } catch (error) {
-    throw error;
-  } finally {
-    clearTimeout(timeout);
-  }
+const fetchAnalysesPage = async (inputs: {
+	dataCenter: DataCenter;
+	studyId: string;
+	offset: number;
+	limit: number;
+	analysisStates?: Set<SongAnalysisState>;
+}): Promise<{ data: SongAnalysis[] } & PaginationMetadata> => {
+	try {
+		const {
+			dataCenter: { centerId, songUrl },
+			studyId,
+			limit,
+			offset,
+			analysisStates,
+		} = inputs;
+		const analysisStatesQueryValue = analysisStates
+			? Array.from(analysisStates).join(',')
+			: SongAnalysisStates.PUBLISHED || SongAnalysisStates.PUBLISHED;
+
+		const queryParams = {
+			analysisStates: analysisStatesQueryValue,
+			limit,
+			offset,
+		};
+
+		const queryString = qs.stringify(queryParams);
+		const analysesUrl = urljoin(songUrl, '/studies', studyId, 'analysis', `paginated?${queryString}`);
+		const response = await fetch(analysesUrl);
+		const data = (await response.json()) as SongResponseAnalysesPage;
+		logger.debug('Successfully retrieved analyses from song', {
+			dataCenter: centerId,
+			studyId,
+			limit,
+			offset,
+			responseCount: data.currentTotalAnalyses,
+			totalCount: data.totalAnalyses,
+		});
+
+		if (!isArray(data.analyses)) {
+			logger.error(`Response from Song paginated analyses endpoint does not contain the expected array of analyses!`, {
+				response: data,
+			});
+			throw new Error(
+				`Response from Song's paginated analyses endpoint does not contain the expected array of analyses`,
+			);
+		}
+
+		return {
+			data: data.analyses,
+			limit,
+			offset,
+			responseCount: data.currentTotalAnalyses,
+			totalCount: data.totalAnalyses,
+		};
+	} catch (e) {
```

Comment:
remind me, had we decided `error` was not the right name for these variables for some reason? even `err` would be more declarative than a single letter

Reply:
Thanks for catching this. Meant to go back and review the error handling before pushing. On that now.